src/git.ts

```diff
@@ -1,43 +1,292 @@
+                    },
+                ]);
+            }
+
+            if(vscode.workspace.getConfiguration('neuropilot').get('permission.gitRemotes', false)) {
+                NEURO.client?.registerActions([
+                    {
+                        name: 'pull_git_commits',
+                        description: 'Pull commits from the remote repository',
+                        schema: {}
+                    },
+                    {
+                        name: 'push_git_commits',
+                        description: 'Push commits to the remote repository',
+                        schema: {
+                            type: 'object',
+                            properties: {
+                                remoteName: { type: 'string' },
+                                branchName: { type: 'string' },
+                                forcePush: { type: 'boolean' },
+                            }
+                        }
+                    }
+                ]);
+
+                if(vscode.workspace.getConfiguration('neuropilot').get('permission.editRemoteData', false)) {
+                    NEURO.client?.registerActions([
+                        {
+                            name: 'add_git_remote',
+                            description: 'Add a new remote to the Git repository',
+                            schema: {
+                                type: 'object',
+                                properties: {
+                                    remoteName: { type: 'string' },
+                                    remoteURL: { type: 'string' },
+                                },
+                                required: ['remoteName', 'remoteURL'],
+                            }
+                        },
+                        {
+                            name: 'remove_git_remote',
+                            description: 'Remove a remote from the Git repository',
+                            schema: {
+                                type: 'object',
+                                properties: {
+                                    remoteName: { type: 'string' },
+                                },
+                                required: ['remoteName'],
+                            }
+                        },
+                        {
+                            name: 'rename_git_remote',
+                            description: 'Rename a remote in the Git repository',
+                            schema: {
+                                type: 'object',
+                                properties: {
+                                    oldRemoteName: { type: 'string' },
+                                    newRemoteName: { type: 'string' },
+                                },
+                                required: ['oldRemoteName', 'newRemoteName'],
+                            }
+                        }
+                    ]);
+                }
+            }
+        }
```

Comment:
Closing bracket for the suggested change above.
```suggestion
        });
```